### ISSUE TITLE
CA1305 reports for nullable types cannot be disabled

### ISSUE THREAD

**Issue by Morilli** [dotnet/roslyn-analyzers#7121]
### Analyzer

**Diagnostic ID**: [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): `Specify IFormatProvider`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.100](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug
In #6730, the CA1305 rule was expanded to also apply to nullable types. However, since that introduction, the analyzer cannot be disabled for any nullable types (or I have not found any way to do so). [This comment](https://github.com/dotnet/roslyn-analyzers/issues/6586#issuecomment-1872277531) in #6586 implies that the analyzer should get disabled, but doesn't.

### Steps To Reproduce
```cs
/// .globalconfig
dotnet_diagnostic.CA1305.severity = error
dotnet_code_quality.CA1305.excluded_symbol_names = T:System.Int32

/// Program.cs
using System;

Random random = new();
int? nullableTestInt = random.Next(2) == 0 ? 20 : null;
int testInt = 20;

// does correctly not show a warning / error
Console.WriteLine(testInt.ToString());

// does result in a build error, even though `T:System.Int32` was excluded from the analyzer
Console.WriteLine(nullableTestInt.ToString());
```

I would expect nullable types to behave the same as their non-nullable counterparts in regards to analyzer exclusion, or alternatively for there to be a different way to also exclude nullable types from the CA1305 analyzer. Both does not seem to be the case currently.

**Comment by mavasani** [comment]
Can you please try `dotnet_code_quality.CA1305.excluded_symbol_names = T:System.Nullable{System.Int32}`?

**Comment by mavasani** [comment]
> Can you please try `dotnet_code_quality.CA1305.excluded_symbol_names = T:System.Nullable{System.Int32}`?

Just verified that this doesn't work either, we should fix the behavior so it does.
